tests/foreman/conftest.py

```diff
@@ -53,35 +53,24 @@ def pytest_report_header(config):
     return messages
 
 
-@pytest.fixture(scope="session")
-def worker_id(request):
-    """Gets the worker ID when running in multi-threading with xdist"""
-    if hasattr(request.config, 'slaveinput'):
-        # return gw+(0..n)
-        return request.config.slaveinput['slaveid']
-    else:
-        return 'master'
-
-
 @pytest.fixture(scope="session")
 def configured_settings():
     if not settings.configured:
         settings.configure()
     return settings
 
 
-@pytest.fixture(autouse=True, scope='module')
+@pytest.fixture(autouse=True, scope='session')
```

Comment:
Simply changing this fixture scope resolved the fact that this was not being executed on the xdist workers.